### ISSUE TITLE
JavaFX not found even after upgrading it via the CLI 

### ISSUE THREAD

**Issue by social-anthrax** [Col-E/Recaf-Launcher#32]
## System Info

MacOS 15.1 (24B83)
Java version: 
```
openjdk 23.0.1 2024-10-15
OpenJDK Runtime Environment Homebrew (build 23.0.1)
OpenJDK 64-Bit Server VM Homebrew (build 23.0.1, mixed mode, sharing)
```
## Issue

Even after installing the latest launcher, JFX still doesn't seem to install correctly. This seems similar to #31 but even after force reinstalling JFX the issue seems to persist.

```
$java -jar ./recaf-cli-0.7.0.jar update-jfx -cf
13:58:46.638 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- Attempting to resolve JFX version in current classpath...
13:58:46.640 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- No JavaFX version class found in the current classpath
13:58:46.646 [main] INFO software.coley.recaf.launcher.LauncherCli -- Java Properties:
 - java.home = /opt/homebrew/Cellar/openjdk/23.0.1/libexec/openjdk.jdk/Contents/Home
 - java.version = 23.0.1
 - java.vm.vendor = Homebrew
 - java.vm.version = 23.0.1
 - javafx.platform = mac-aarch64
 - javafx.version.cached = 24-ea+15
 - javafx.version.runtime = -1
 - os.arch = aarch64
 - os.bitness = 64
 - os.name = Mac OS X
 - os.processors = 10
 - os.version = 15.1
 - recaf.version = 4.0.0-SNAPSHOT

13:58:46.729 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- JFX cache:
 - Files: 9974706
 - Size:  9974706
13:58:46.730 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- Clearing dependency cache
13:58:46.731 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- Deleting dependency javafx-base-24-ea+15-mac-aarch64.jar
13:58:46.731 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- Deleting dependency javafx-graphics-24-ea+15-mac-aarch64.jar
13:58:46.732 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- Deleting dependency javafx-media-24-ea+15-mac-aarch64.jar
13:58:46.732 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- Deleting dependency javafx-controls-24-ea+15-mac-aarch64.jar
13:58:47.674 [main] INFO software.coley.recaf.launcher.LauncherCli -- Updated to JavaFX '24-ea+15'
```

```
$ java -jar ./recaf-cli-0.7.0.jar auto
13:56:49.415 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- Attempting to resolve JFX version in current classpath...
13:56:49.417 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- No JavaFX version class found in the current classpath
13:56:49.423 [main] INFO software.coley.recaf.launcher.LauncherCli -- Java Properties:
 - java.home = /opt/homebrew/Cellar/openjdk/23.0.1/libexec/openjdk.jdk/Contents/Home
 - java.version = 23.0.1
 - java.vm.vendor = Homebrew
 - java.vm.version = 23.0.1
 - javafx.platform = mac-aarch64
 - javafx.version.cached = 24-ea+15
 - javafx.version.runtime = -1
 - os.arch = aarch64
 - os.bitness = 64
 - os.name = Mac OS X
 - os.processors = 10
 - os.version = 15.1
 - recaf.version = 4.0.0-SNAPSHOT

13:56:49.505 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- JFX cache:
 - Files: 9974706
 - Size:  9974706
13:56:49.775 [main] INFO software.coley.recaf.launcher.LauncherCli -- Current JavaFX is up-to-date: '24-ea+15'
13:56:50.021 [main] DEBUG software.coley.recaf.launcher.LauncherCli -- Looking in '/Users/lyra/Library/Application Support/Recaf' for Recaf/dependencies...
13:56:50.028 [main] ERROR software.coley.recaf.launcher.LauncherCli -- The current runtime bundles JavaFX that is too old (null).
- Try running with a JDK that has an up-to-date JavaFX bundled
- Or use a JDK that does not bundle JavaFX and let the launcher grab it for you
```

## Additional info 
Contents of `/Users/<user>/Library/Application\ Support/Recaf/dependencies`
```
javafx-base-24-ea+15-mac-aarch64.jar
javafx-controls-24-ea+15-mac-aarch64.jar
javafx-graphics-24-ea+15-mac-aarch64.jar
javafx-media-24-ea+15-mac-aarch64.jar
```

**Comment by Col-E** [comment]
> The current runtime bundles JavaFX that is too old (null)

Welp, thats certainly not intended.

**Comment by Col-E** [comment]
This should be addressed in 0.8.0 with the changes in how we ensure FX is loaded into the child process
